### PR TITLE
feat: Login flow v2

### PR DESCRIPTION
**Pull request by blvdek**
Changes proposed in this pull request:

 * Implement Login flow v2 for standard Nextcloud clients.
 * Allow user to initialize standard Nextcloud clients without providing credentials so that the user can log in using Login flow v2.

### CODE REVIEW

**Comment by bigcat88** [comment]
and for tests(test_init_poll, test_init_poll_async)  as this stuff is not for AppAPI:

`nc_any` should be `nc_client` and `anc_any` should be `anc_client`

```
def test_init_poll(nc_client):
    lf = nc_client.loginflow_v2.init()
    ...
```    

this help to not trigger these tests for `NextcloudApp` class.

File: nc_py_api/loginflow_v2.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} login={self.login_name} app_password={self.app_password}>"
+
+
+class _LoginFlowV2API:
+    """Class implementing Nextcloud Login flow v2."""
+
+    _ep_init: str = "/index.php/login/v2"
+    _ep_poll: str = "/login/v2/poll"
```
Comment (bigcat88) [inline nc_py_api/loginflow_v2.py:78]: why here is without "index.php" ?

File: nc_py_api/loginflow_v2.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            self._session.init_adapter(restart=True)
+            self._session.init_adapter_dav(restart=True)
+        return r_model
+
+
+class _AsyncLoginFlowV2API:
+    """Class implementing Async Nextcloud Login flow v2."""
+
+    _ep_init: str = "/index.php/login/v2"
+    _ep_poll: str = "/login/v2/poll"
```
Comment (bigcat88) [inline nc_py_api/loginflow_v2.py:120]: the same, in my dev setup, without "index.php" it does not work..

**Comment by bigcat88** [comment]
As I understand this should be used like this:

```python3
import nc_py_api


if __name__ == "__main__":
    nc = nc_py_api.Nextcloud(nextcloud_url="http://stable29.local")
    r = nc.loginflow_v2.init()
    print(r)  # here open link in browser and agree to connect
    r2 = nc.loginflow_v2.poll(r.token)
    print(r2)
    print(nc.capabilities)
    print(nc.files.listdir())
```

---

Thank you for your pull request and interest in Nextcloud and nc_py_api, after changing the URL I will gladly merge it. :)

**Comment by bigcat88** [comment]
Thanks again, good job :)

**Comment by blvdek** [comment]
Thank you too

**Pull request merged**